### PR TITLE
dai: remove dai frame format configuration.

### PR DESCRIPTION
**Pull request by btian1**
frame_fmt was later overwritten by ipc_config.framefmt, so the previous assignment based on cfg does not make sense anymore.

Signed-off-by: Baofeng Tian <baofeng.tian@intel.com>

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
@abonislawski @juimonen pls review.

**Comment by lgirdwood** [comment]
ping @abonislawski @juimonen

**Review by lgirdwood — APPROVED** [review]

File: src/audio/dai-zephyr.c [hunk 1]
```diff
@@ -415,8 +401,13 @@ static int dai_comp_hw_params(struct comp_dev *dev, struct sof_ipc_stream_params
 static int dai_verify_params(struct comp_dev *dev, struct sof_ipc_stream_params *params)
 {
 	struct sof_ipc_stream_params hw_params;
+	int ret;
 
-	dai_comp_get_hw_params(dev, &hw_params, params->direction);
+	ret = dai_comp_get_hw_params(dev, &hw_params, params->direction);
```
Comment (ranj063) [inline src/audio/dai-zephyr.c:406]: why not remove dai_get_hw_params() altogether and just set the params here?

Comment (btian1) [reply src/audio/dai-zephyr.c:406]: yes, this is also achievable, if there is no other concern and history, I can remove dai_get_hw_params totally.
@ranj063 @lgirdwood

Comment (ranj063) [reply src/audio/dai-zephyr.c:406]: it is a static function. so I dont see any concern with removing it. That function doesnt really get anything but rather just set things. So might as well do those things directly here.

Comment (btian1) [reply src/audio/dai-zephyr.c:407]: ok, done

File: src/audio/dai-zephyr.c [hunk 2]
```diff
@@ -384,17 +356,17 @@ static int dai_comp_get_hw_params(struct comp_dev *dev,
 				  int dir)
 {
 	struct dai_data *dd = comp_get_drvdata(dev);
-	int ret;
+	const struct dai_config *cfg = dai_config_get(dd->dai->dev, dir);
 
 	comp_dbg(dev, "dai_hw_params()");
 
 	/* fetching hw dai stream params */
-	ret = dai_get_hw_params(dd->dai, params, dir);
```
Comment (ranj063) [inline src/audio/dai-zephyr.c:392]: the comment above needs to be removed/changed as well?

Comment (btian1) [reply src/audio/dai-zephyr.c:392]: ok

**Comment by btian1** [comment]
regression bug caused CI failure: https://github.com/thesofproject/sof/issues/6947
not my PR issue, no other issue found.

**Review by juimonen — APPROVED** [review]

**Pull request merged**